assets/js/components/legacy-notifications/util.js

```diff
@@ -87,10 +88,15 @@ const removeDismissed = ( notifications ) => {
 		return notifications;
 	}
 
-	return notifications.filter( ( notification ) => {
-		const dismissed = getCache( `notification::dismissed::${ notification.id }` );
+	const promises = await notifications.map( ( notification ) => getItem( `notification::dismissed::${ notification.id }` ) );
+	const allNotifications = await Promise.all( promises );
+
+	const notDismissed = notifications.filter( ( _, index ) => {
+		const dismissed = allNotifications[ index ];
```

Comment:
This is not quite accurate as `getItem` will always resolve an object: https://github.com/google/site-kit-wp/blob/3db19cbeca41145aefc6c6c3383ee9b66e8b4121/assets/js/googlesitekit/api/cache.js#L184-L196

```suggestion
		const dismissed = allNotifications[ index ].cacheHit;
```